### PR TITLE
Remove MSTest dependency from interaction tests

### PR DESCRIPTION
**Pull request by llongley**
We currently double-compile all of our tests, once against MSTest for VS Test Explorer integration, and again against TAEF for command-line test execution.  The latest version of TAEF supports VS Test Explorer integration, however, so these changes get us a lot of the way to being able to remove MSTest entirely and just use TAEF for everything.

VS Test Explorer is having trouble finding our AppX-based tests when using TAEF, so for now I've left those alone while I figure that out, but these changes remove MSTest and get us on TAEF fully for all our DLL-based tests.

I also left in some test code to enable running API tests in the test app for easier debugging, and I fixed up the inner-loop solution to work again.

### CODE REVIEW

File: nuget.config [hunk 1]
```diff
@@ -9,6 +9,7 @@
     <add key="localpackages" value="localpackages" />
   </packageSources>
   <config>
-    <add key="repositoryPath" value="$\..\packages" />
+    <add key="repositoryPath" value="packages" />
+    <add key="globalPackagesFolder" value="packages" />
```
Comment (kmahone) [inline nuget.config:13]: What is the purpose of these changes?

File: dev/GeneratedExperimental/TitleBar.properties.cpp [hunk 2]
```diff
@@ -92,7 +92,7 @@ void TitleBarProperties::EnsureProperties()
                 winrt::name_of<winrt::TitleBar>(),
                 false /* isAttached */,
                 ValueHelper<winrt::hstring>::BoxedDefaultValue(),
-                nullptr);
+                winrt::PropertyChangedCallback(&OnTitlePropertyChanged));
```
Comment (kmahone) [inline dev/GeneratedExperimental/TitleBar.properties.cpp:95]: Are the changes in these files expected?

File: test/MUXControls.Test/TestAssembly.cs [hunk 3]
```diff
@@ -37,8 +32,8 @@ namespace Windows.UI.Xaml.Tests.MUXControls.InteractionTests.Infra
     public class TestAssembly
     {
         [AssemblyInitialize]
-        [TestProperty("CoreClrProfile", ".NETCoreApp2.1")]
-        [TestProperty("RunFixtureAs:Assembly", "ElevatedUserOrSystem")]
+        [TestProperty("CoreClrProfile", ".")]
+        //[TestProperty("RunFixtureAs:Assembly", "ElevatedUserOrSystem")]
```
Comment (kmahone) [inline test/MUXControls.Test/TestAssembly.cs:36]: Do we still need this?

File: build/AzurePipelinesTemplates/MUX-InstallDotNetSDK-Steps.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+parameters:
+  sdkVersion: 5.0.404
+
+steps:
+  - powershell: Invoke-WebRequest "https://dotnet.microsoft.com/download/dotnet/scripts/v1/dotnet-install.ps1" -OutFile "$(Agent.TempDirectory)\dotnet-install.ps1"
```
Comment (kmahone) [inline build/AzurePipelinesTemplates/MUX-InstallDotNetSDK-Steps.yml:5]: Can we use the UseDotNet@2 task instead here?

See [https://github.com/microsoft/microsoft-ui-xaml/blob/f9827bb2086f0b86d5a8a3caa264527cf6acf472/build/AzurePipelinesTemplates/MUX-BuildProject-Steps.yml#L18](https://github.com/microsoft/microsoft-ui-xaml/blob/f9827bb2086f0b86d5a8a3caa264527cf6acf472/build/AzurePipelinesTemplates/MUX-BuildProject-Steps.yml#L18)

File: build/Helix/PrepareHelixPayload.ps1 [hunk 5]
```diff
@@ -39,11 +40,11 @@ Write-Host "netCoreAppVer = $netCoreAppVer"
...
 Copy-Item "$nugetPackagesDir\microsoft.windows.apps.test.$mitaVer\lib\netcoreapp2.1\*.dll" $payloadDir
-Copy-Item "$nugetPackagesDir\taef.redist.wlk.$taefVer\build\Binaries\$Platform\*" $payloadDir
-Copy-Item "$nugetPackagesDir\taef.redist.wlk.$taefVer\build\Binaries\$Platform\CoreClr\*" $payloadDir
-New-Item -ItemType Directory -Force -Path "$payloadDir\.NETCoreApp2.1\"
-Copy-Item "$nugetPackagesDir\runtime.win-$Platform.microsoft.netcore.app.$netCoreAppVer\runtimes\win-$Platform\lib\netcoreapp2.1\*" "$payloadDir\.NETCoreApp2.1\"
-Copy-Item "$nugetPackagesDir\runtime.win-$Platform.microsoft.netcore.app.$netCoreAppVer\runtimes\win-$Platform\native\*" "$payloadDir\.NETCoreApp2.1\"
+Copy-Item "$nugetPackagesDir\Microsoft.Taef.$taefVer\build\Binaries\$Platform\*" $payloadDir
+Copy-Item "$nugetPackagesDir\Microsoft.Taef.$taefVer\build\Binaries\$Platform\netstandard2.0\*" $payloadDir
+Copy-Item "$nugetPackagesDir\Microsoft.NETCore.App.Runtime.win-$Platform.$netCoreAppVer\runtimes\win-$Platform\lib\net5.0\*" $payloadDir
+Copy-Item "$nugetPackagesDir\Microsoft.NETCore.App.Runtime.win-$Platform.$netCoreAppVer\runtimes\win-$Platform\native\*" $payloadDir
```
Comment (kmahone) [inline build/Helix/PrepareHelixPayload.ps1:46]: >$payloadDir [](http://example.com/codeflow?start=121&length=11)

Previously, I ran into issues doing this. When the .net runtime binaries are copied into the root directory instead of into ".NetCoreApp2.1" I hit an issue where the powershell scripts that we run would try to load these dlls which was causing failures.

So it is probably best if we continue to copy the runtime dlls into a subdir.

File: global.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (kmahone) [inline global.json:1]: Are we completely off .net Core 2/3 now? Or are we still using it anywhere?

If we are fully onto .net 5, can you update this Issue:
Update test code to .net 6 #6487

We should still update to .net 6 at some point, but now that we are on .net 5, it is much less urgent.

**Comment by kmahone** [comment]
> <Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">

Do we still need the MSTest version of this app?

---
Refers to: test/MUXControlsTestApp/MSTest/MUXControlsTestApp.csproj:3 in eb47c10. [](commit_id = eb47c102b3358c46b43b187a998e60af4e78a18c, deletion_comment = False)

File: test/testinfra/MUXTestInfra/TestAppInstallHelper.cs [hunk 7]
```diff
@@ -137,13 +135,58 @@ FileInfo FirstFileWithExtension(params string[] extensions)
...
+        /// </summary>
+        public static bool InstallTestAppFromDirectoryIfNeeded(string testAppDirectory, string packageFamilyName)
+        {
+            if (!TestAppxInstalled.Contains(packageFamilyName))
+            {
+                var createAppxDirectoryScript = Path.Combine(testAppDirectory, "CreateAppxDirectory.ps1");
+
+                if (File.Exists(createAppxDirectoryScript))
+                {
+                    // To install from a directory, we have a script that does two things: first, it runs an MSBuild script
```
Comment (kmahone) [inline test/testinfra/MUXTestInfra/TestAppInstallHelper.cs:159]: Instead of invoking MSBuild from the test code (which is a bit topsy-turvy) can we make the CreateAppxPackage Target run as part of the main build?

File: test/testinfra/MUXTestInfra/Infra/TestEnvironment.cs [hunk 8]
```diff
@@ -105,57 +115,77 @@ public static TestApplicationInfo MUXControlsTestApp
         {
             get
             {
-#if USING_TAEF
-                string testAppName = "MUXControlsTestApp_8wekyb3d8bbwe!taef.executionengine.universal.App";
+                // When building inside Visual Studio, we'll install the test app from a directory to avoid needing to build and install an AppX,
+                // since that mimics how F5 deployment works from Visual Studio and is a lot faster.
+                // When building from a command line, we don't have Visual Studio integration, so we'll build and install an AppX in that case.
+#if INSTALL_FROM_APPX
```
Comment (kmahone) [inline test/testinfra/MUXTestInfra/Infra/TestEnvironment.cs:121]: >INSTALL_FROM_APPX [](http://example.com/codeflow?start=4&length=17)

This makes me cringe a little bit, since it violates the principle of least surprise - I would expect the output of building the dll with msbuild to match building it with VS. I wonder is there another approach we could take?

**Comment by kmahone** [comment]
>             Log.Comment("Enable side loading apps");

Does this require us to run VS elevated if we want to run tests?

---
Refers to: test/testinfra/MUXTestInfra/Infra/TestEnvironment.cs:264 in eb47c10. [](commit_id = eb47c102b3358c46b43b187a998e60af4e78a18c, deletion_comment = False)

**Review by kmahone — CHANGES_REQUESTED** [review]
:clock1:

**Comment by gabbybilka** [comment]
Closing as stale and focusing on WinUI 3.
